src/main/java/com/connexta/store/poller/TransformStatusPoller.java

```diff
@@ -0,0 +1,281 @@
+        log.warn(
+            "Bad addMetadata request sent to store for dataset {}. Request will not be retried. \n{}",
+            datasetId,
+            addMetadataRequest);
+        return false;
+      } else {
+        return true;
+      }
+    } else {
+      log.debug("No response received from add metadata request for dataset {}.", datasetId);
+      return true;
+    }
+  }
+
+  /** @return {@code true} if the quarantine should be retried, {@code false} otherwise */
+  private boolean quarantine(String datasetId, List<MetadataInformation> metadataInformations) {
+    QuarantineRequest quarantineRequest = new QuarantineRequest();
+    for (MetadataInformation metadataInformation : metadataInformations) {
+      quarantineRequest.addMetadataInfo(
+          new MetadataInfo()
+              .metadataType(metadataInformation.getMetadataType())
+              .location(metadataInformation.getLocation()));
+    }
+
+    ClientResponse response =
+        storeWebClient
+            .put()
+            .uri(
+                UriComponentsBuilder.fromPath(StoreController.QUARANTINE_URL_TEMPALTE)
+                    .build(datasetId))
+            .body(BodyInserters.fromValue(quarantineRequest))
+            .exchange()
+            .block();
+
+    if (response != null) {
+      final HttpStatus httpStatus = response.statusCode();
+      if (HttpStatus.OK.equals(httpStatus)) {
+        deleteTransformationStatus(datasetId);
+        return false;
+      } else if (HttpStatus.NOT_FOUND.equals(httpStatus)) {
+        log.debug("Failed to quarantine dataset {} because it no longer exists.", datasetId);
+        deleteTransformationStatus(datasetId);
+        return false;
+      } else if (HttpStatus.BAD_REQUEST.equals(httpStatus)) {
+        log.warn(
+            "Bad quarantine request sent to store for dataset {}. Request will not be retried. \n{}",
+            datasetId,
+            quarantineRequest);
+        return false;
+      } else {
+        return true;
+      }
+    } else {
+      log.debug("No response received from quarantine request for dataset {}.", datasetId);
+      return true;
+    }
+  }
+
+  /**
+   * This method is best effort. Whether or not this method fails to remove the transformation
+   * status from transform, we will not let it affect whether or not the TransformStatusTask is
+   * tried again.
+   */
+  private void deleteTransformationStatus(String datasetId) {
+    ClientResponse response =
+        transformWebClient
+            .delete()
+            .uri(
+                UriComponentsBuilder.fromPath(
+                        TransformConfiguration.TRANSFORM_STATUS_DELETE_TEMPLATE)
+                    .build(datasetId)
+                    .toASCIIString())
+            .exchange()
+            .block();
+
+    if (response != null) {
+      final HttpStatus httpStatus = response.statusCode();
+      if (HttpStatus.NO_CONTENT.equals(httpStatus)) {
```

Comment:
❓ Any reason to have all this logic if this is a best effort scenario with no retries?

Reply:
Mostly for logging, but I could be convinced to delete it cause I decided not to test these cases since they are pretty untestable.

Reply:
And transform should be logging all these cases.

Reply:
Yeah i'll just remove all this logic and log the attempt.